cylc/flow/data_store_mgr.py

```diff
@@ -320,16 +345,24 @@ def apply_delta(key, delta, data):
             for field in CLEAR_FIELD_MAP[key]:
                 if field in field_set or delta.updated.states_updated:
                     data[key].ClearField(field)
+
+            # Apply the updated delta
             data[key].MergeFrom(delta.updated)
+
+            # Deque (i.e. length restrict) selected fields
+            for field_name, max_len in DEQUE_FIELD_MAP.get(key, {}).items():
+                if field_name in field_set:
+                    lst = getattr(data[key], field_name)
+                    while len(lst) > max_len:
+                        lst.pop(0)
         else:
             for element in delta.updated:
                 try:
                     data_element = data[key][element.id]
                     # Clear fields that require overwrite with delta
-                    if CLEAR_FIELD_MAP[key]:
```

Comment:
The reason I had this is to avoid iterating over the fields unnecessarily if there are no fields flagged for clearing for this type/key.